cmd/scaffold.go

```diff
@@ -526,8 +532,20 @@ func (fnb *FlowNodeBuilder) initState() {
 	// 2) the node id is a new one for a new spork, but the bootstrap data has not been updated.
 	myID, err := flow.HexStringToIdentifier(fnb.BaseConfig.nodeIDHex)
 	fnb.MustNot(err).Msg("could not parse node identifier")
-	self, err := fnb.State.Final().Identity(myID)
-	fnb.MustNot(err).Msgf("node identity not found in the identity list of the finalized state: %v", myID)
+
+	var self *flow.Identity
+	if fnb.unstaked {
+		self = &flow.Identity{
+			NodeID:        myID,
+			NetworkPubKey: fnb.networkKey.PublicKey(),
+			StakingPubKey: nil,             // no staking key needed for the unstaked node
+			Role:          flow.RoleAccess, // unstaked node can only run as an access node
+			Address:       fnb.BaseConfig.bindAddr,
+		}
+	} else {
+		self, err = fnb.State.Final().Identity(myID)
+		fnb.MustNot(err).Msgf("node identity not found in the identity list of the finalized state: %v", myID)
+	}
 
 	// Verify that my role (as given in the configuration) is consistent with the protocol state.
 	// We enforce this strictly for MainNet. For other networks (e.g. TestNet or BenchNet), we
```

Comment:
this existing check, will also catch the error if the `Role` in the node-info.json is anything other than `access`